addons/crm_mail_plugin/tests/test_crm_mail_plugin.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import json
+
+from odoo.addons.mail_plugin.tests.common import TestMailPluginControllerCommon
+
+
+class TestCrmMailPlugin(TestMailPluginControllerCommon):
+    def test_get_contact_data(self):
+        """Check that the leads section is not visible if the user has not access to crm.lead."""
+        partner = self.env["res.partner"].create({"name": "Test"})
+
+        data = {
+            "id": 0,
+            "jsonrpc": "2.0",
+            "method": "call",
+            "params": {"partner_id": partner.id},
+        }
+
+        with self.mock_auth_method():
```

Comment:
Can't you use those as method decorators?
I find that a little more elegant.

Reply:
it would be possible if `mock_auth_method` was a function and not a method (but it needs `self`, so it's not possible)

Reply:
Could we make it take the user to replace as an argument then?
If really not possible, it's fine to keep it like this though.

Reply:
@awa-odoo No it's possible :) (we can access the environment thanks to the request object, to retrieve the user_id)

Tell me what you think